docs/content/ruby/guides/data_filtering_next/index.md

```diff
@@ -1,104 +1,68 @@
+It currently has four fields: `root`, `relations`, `conditions`, and `types`.
 
-The other thing you have to provide to use data filtering is type information
-for registered classes. This lets Oso know what the types of an object's fields
-are. Oso needs this information to handle specializers and other things in the
-policy when we don't have a concrete resource. The fields are a 
-{{% exampleGet "map" %}} from field name to type.
+The `{{% exampleGet "buildQuery" %}}` method's job is to create a query over the
+set of tables implied by the `relations`, for those records matching the
+`conditions`. The `conditions` represent an `OR` of `AND`s, so a record matches if
+for any top level list every condition triple it contains is true for that record.
 
-## Relations
+##### Filters
 
-Often you need data that is not contained on the object to make
-authorization decisions. This comes up when the role required to
-do something is implied by a role on it's parent object. For instance,
-you want to check the organization for a repository but that data isn't
-embedded on the repository object. You can add a `Relation` type to the type
-definition that states how the other resource is related to this one. Then
-you can access this field in the policy like any other field and it will
-fetch the data when it needs it (via the query functions).
+The four attributes of a `Filter` are:
+- `root`: the class of the queried resource
+- `relations`: a list of `[from_class, rel_name, to_class]` triples,
+  with values corresponding to your globally registered relations. We apologize
+  for the number of different things we designate a `relation`. This will be
+  addressed in an upcoming release.
+- `conditions`: a list of lists of `[lhs, op, rhs]` triples, with `op` in
+  `%w[Eq Neq In Nin]`, and `lhs` and `rhs` being `Projection` objects having a
+  `source` (a resource class) and a `field` (an optional string representing a
+  column name, or `nil` indicating "object identity" according to your case
+  (for example, compare by a primary key).
+- `types`: a hash whose keys are class objects and their names and registered
+  aliases, and whose values are Oso-defined type information needed to understand
+  the `Filter`'s `relations`. This attribute may be removed in an
+  upcoming release.
 
-`Relation`s are a special type that tells Oso how one Class is related to
-another. They specify what the related type is and how it's related.
-
-- `kind` is either "one" or "many". "one" means there is one related object and
-      "many" means there is a list of related objects.
-- `other_type` is the class of the related objects.
-- `my_field` Is the field on this object that matches `other_field`.
-- `other_field` Is the field on the other object that matches `this_field`.
-
-The `my_field` / `other_field` relationship is similar to a foreign key. It lets Oso
-know what fields to match up with building a query for the other type.
 
 ## Example
```

Comment:
This example doesn't really show how to implement an Adapter since it just uses the pre-built active record adapter.